src/ssh-compute.ts

```diff
@@ -0,0 +1,189 @@
+ */
+'use strict';
+
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import * as toolCache from '@actions/tool-cache';
+import * as setupGcloud from '@google-github-actions/setup-cloud-sdk';
+
+import path from 'path';
+import { promises as fs } from 'fs';
+
+import { SSH_PUBLIC_KEY_FILENAME, SSH_PRIVATE_KEY_FILENAME } from './const';
+
+export const GCLOUD_METRICS_ENV_VAR = 'CLOUDSDK_METRICS_ENVIRONMENT';
+export const GCLOUD_METRICS_LABEL = 'github-actions-ssh-compute';
+
+export function parseFlags(flags: string): RegExpMatchArray {
+  return flags.match(/(".*?"|[^"\s=]+)+(?=\s*|\s*$)/g)!; // Split on space or "=" if not in quotes
+}
+
+/**
+ * Executes the main action. It includes the main business logic and is the
+ * primary entry point. It is documented inline.
+ */
+export async function run(): Promise<void> {
+  core.exportVariable(GCLOUD_METRICS_ENV_VAR, GCLOUD_METRICS_LABEL);
+  try {
+    // Get inputs
+    // Core inputs
+    let instanceName = core.getInput('instance_name');
+    const zone = core.getInput('zone');
+    const user = core.getInput('user');
+    const ssh_public_key = core.getInput('ssh_public_key');
+    const ssh_private_key = core.getInput('ssh_private_key');
+    const ssh_keys_folder = core.getInput('ssh_keys_folder');
+    const container = core.getInput('container');
+    const sshArgs = core.getInput('ssh_args');
+    let command = core.getInput('command');
+    const script = core.getInput('script');
+    const projectId = core.getInput('project_id');
+    let gcloudVersion = core.getInput('gcloud_version');
+
+    // Flags
+    const flags = core.getInput('flags');
+    const installBeta = true; // Flag for installing gcloud beta components
+    let cmd;
+
+    if (command && script) {
+      throw new Error('Both `command` and `entrypoint` inputs set - Please select one.');
+    }
+
+    if (user) {
+      instanceName = `${user}@${instanceName}`;
+    }
+
+    cmd = [
+      'compute',
+      'ssh',
+      instanceName,
+      '--zone',
+      zone,
+      '--ssh-key-file',
+      `${ssh_keys_folder}/${SSH_PRIVATE_KEY_FILENAME}`,
+      '--quiet', // we need to ignore prompts from console
+      '--tunnel-through-iap',
+    ];
+
+    // Save public and private ssh keys to the temp folder
+    await fs.mkdir(ssh_keys_folder, { recursive: true });
+    await fs.writeFile(`${ssh_keys_folder}/${SSH_PUBLIC_KEY_FILENAME}`, ssh_public_key, {
+      mode: 0o600,
+    });
+    await fs.writeFile(`${ssh_keys_folder}/${SSH_PRIVATE_KEY_FILENAME}`, '', { mode: 0o600 });
+    for (const key of ssh_private_key.split(/(?=-----BEGIN)/)) {
+      await fs.appendFile(`${ssh_keys_folder}/${SSH_PRIVATE_KEY_FILENAME}`, key.trim() + '\n');
+    }
+
+    if (container) {
+      cmd.push('--container', container);
+    }
+
+    if (flags) {
+      const flagList = parseFlags(flags);
+      if (flagList) cmd = cmd.concat(flagList);
+    }
+
+    if (script) {
+      try {
```

Comment:
This is a race condition and general anti-pattern. Always read the file, then check for an error. Checking if the file exists doesn't guarantee it will exist when you read it.